src/Legacy/DynamoRevitVersionSelector/ProductLookUp.cs

```diff
@@ -0,0 +1,345 @@
+        /// </summary>
+        /// <param name="installPath">Install location</param>
+        /// <returns>Full path for core file</returns>
+        string GetCoreFilePathFromInstallation(string installPath);
+
+        /// <summary>
+        /// Gets file version info from the given path.
+        /// </summary>
+        /// <param name="filePath">File path</param>
+        /// <returns>Version info as Tuple</returns>
+        Tuple<int, int, int, int> GetVersionInfoFromFile(string filePath);
+
+    }
+
+    /// <summary>
+    /// Represents collection of installed productss
+    /// </summary>
+    public interface IProductCollection
+    {
+        /// <summary>
+        /// Gets list of installed products
+        /// </summary>
+        IEnumerable<IInstalledProduct> Products { get; }
+
+        /// <summary>
+        /// Gets latest product from installation
+        /// </summary>
+        /// <returns>Installed product</returns>
+        IInstalledProduct GetLatestProduct();
+
+        /// <summary>
+        /// Gets all installed product on the system using the given lookUp 
+        /// and initializes itself.
+        /// </summary>
+        /// <param name="lookUp">LookUp interface</param>
+        void LookUpAndInitProducts(IProductLookUp lookUp);
+    }
+
+    /// <summary>
+    /// Implements basic look up algorithm to get product installations using registry keys
+    /// </summary>
+    public class InstalledProductLookUp : IProductLookUp
+    {
+        const string REG_KEY64 = @"SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\";
+        const string REG_KEY32 = @"SOFTWARE\Wow6432Node\Microsoft\Windows\CurrentVersion\Uninstall\";
+
+        /// <summary>
+        /// Product name for lookup
+        /// </summary>
+        public string LookUpProductName { get; private set; }
+        private readonly string fileLookup;
+
+        static RegistryKey OpenKey(string key)
+        {
+            var regKey = RegistryKey.OpenBaseKey(RegistryHive.LocalMachine, RegistryView.Registry64);
+            return regKey.OpenSubKey(key);
+        }
+
+        static string GetInstallLocation(RegistryKey key)
+        {
+            if (key != null)
+                return key.GetValue("InstallLocation") as string;
+
+            return string.Empty;
+        }
+
+        static string GetDisplayName(RegistryKey key)
+        {
+            if (key != null)
+                return key.GetValue("DisplayName") as string;
+
+            return string.Empty;
+        }
+
+        /// <summary>
+        /// Implements a product look up algorithm based on registry key.
+        /// </summary>
+        /// <param name="lookUpName">Product name to lookup</param>
+        /// <param name="fileLookup">file name pattern to lookup</param>
+        public InstalledProductLookUp(string lookUpName, string fileLookup)
+        {
+            this.LookUpProductName = lookUpName;
+            this.fileLookup = fileLookup;
+        }
```

Comment:
For Dynamo lookup name is Dynamo and the file lookup name is DynamoCore.dll to validate the Dynamo installation.